docs/src/ring_interface.md

```diff
@@ -996,19 +1007,51 @@ function constant_polynomial_ring(R::Ring, cached::Bool=true)
    T = elem_type(R)
    return ConstPolyRing{T}(R, cached)
 end
+
+# output
+
+constant_polynomial_ring (generic function with 2 methods)
 ```
 
 The above implementation of `constant_polynomial_ring` may be tested as follows.
 
-```julia
+```jldoctest ConstPoly; filter = r".*"
 using Test
 include(joinpath(pathof(AbstractAlgebra), "..", "..", "test", "Rings-conformance-tests.jl"))
 
-S, _ = polynomial_ring(QQ, :x)
+function test_elem(R::ConstPolyRing{elem_type(ZZ)})
+   n = rand(1:999)
+   return R(rand(-n:n))
+end
+
+test_Ring_interface(constant_polynomial_ring(ZZ))
 
-function test_elem(R::ConstPolyRing{elem_type(S)})
-   return R(rand(base_ring(R), 1:6, -999:999))
+# output
+
+```
+
+Note that we only showed a minimal implementation of the ring interface.
+Additional interfaces exists, e.g. for Euclidean rings. Additional interface
+usually require implementing additional methods, and in some cases we also
+provide additional conformance tests. In this case, just one necessary
+method is missing.
+
+```jldoctest ConstPoly
+function Base.divrem(a::ConstPoly{elem_type(ZZ)}, b::ConstPoly{elem_type(ZZ)})
+   check_parent(a, b)
+   q, r = AbstractAlgebra.divrem(a.c, b.c)
+   return parent(a)(q), parent(a)(r)
 end
 
-test_Ring_interface(constant_polynomial_ring(S))
+# output
+
+```
+
+We can test it like this.
+
+```jldoctest ConstPoly; filter = r".*"
```

Comment:
So this filter (and the one above in line 1018) is really meant to discard all output. This is because the output in `test_EuclideanRing_interface` comes from `@testset`. But there is only output if this is the top-most `@testset`. But it turns out that depending on how doctests are tested, they may either all be run in a single encompassing `@testset`, or not. Thus in one code path Documenter will complain here if I provide example output, and in the other if I don't provide it. The filter is the only one out.

Luckily this does not impeded the effectiveness of this test: If `test_EuclideanRing_interface` reports a failure, it still gets reported at least in the case with the encompassing `@testset`. I verified this by deliberately breaking one of the methods and checking that `doctest` reports this failure.